File: B67156/B63817/Laboratorios/19-03/src/main/java/cr/ac/ucr/ie/app/domain/Departamento.java
package cr.ac.ucr.ie.app.domain;

public class Departamento
{	
	private int cod;
	private String nombre;
	
	public Departamento(int cod, String nombre)
	{
		if(cod <= 0) throw new RuntimeException("El id debe ser mayor a 0.");
		if(nombre == null || nombre.trim().equals("")) throw new RuntimeException("El nombre es requerido");
		
		this.cod = cod;
		this.nombre = nombre;
	}

	public int getCod() {
		return cod;
	}

	public void setCod(int cod) {
		if(nombre == null || nombre.trim().equals("")) throw new RuntimeException("El nombre es requerido");
		this.cod = cod;
	}

	public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		if(cod <= 0) throw new RuntimeException("El id debe ser mayor a 0.");
		this.nombre = nombre;
	}
	
	
	
}

File: B40894/Examen2/model/Domain/Departamento.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace model.Domain
{
    public class Departamento
    {
        


        public Departamento()
        {
          

        }


        public Departamento(int idDepartamento, string nombre, Departamento jefatura)
        {
            IdDepartamento = idDepartamento;
            Nombre = nombre;
            Marcado = false;
            Jefatura = jefatura;
            
        }


        private Departamento jefatura;
        public Departamento Jefatura
        {
            get
            {
                return jefatura;
            }
            set
            {

                jefatura = value;
            }
        }


        private string nombre;
        public string Nombre
        {
            get
            {
                return nombre;
            }
            set
            {
                if (string.IsNullOrEmpty(value)) throw new Exception("Nombre es requerido.");
                if (string.IsNullOrWhiteSpace(value)) throw new Exception("No puede ingresar solo espacios");
                if (Jefatura != null) {
                    if (value.Equals(Jefatura.nombre)) throw new Exception("Nombre de departamento ya existe, no puede llamarse igual a su jefatura");
                } 
                nombre = value;
            }
        }

        private int idDepartamento;
        public int IdDepartamento
        {
            get
            {
                return idDepartamento;
            }
            set
            {
              if (value <= 0) throw new Exception("El numero debe ser positivo");
                idDepartamento = value;
            }
        }

        private bool marcado;
        public bool Marcado
        {
            get
            {
                return marcado;
            }
            set
            {
                marcado = value;
            }
        }

    }
}
